### ISSUE TITLE
yolov5s implemented on Jetson nano

### ISSUE THREAD

**Issue by MinnanZhou** [wang-xinyu/tensorrtx#569]
I successfully converted my trained model to a tensorrt engine based on your nice code on yolov5. The inference time with 416 size input is about 37ms.
By the way, how to cite your code? I hope to use it in my thesis. Thanks a lot!

**Comment by wang-xinyu** [comment]
You can go with latex format like
```
@misc{tensorrtx,
  author = {Xinyu Wang, etc.},
  title = {tensorrtx},
  year = {2021},
  publisher = {GitHub},
  journal = {GitHub repository},
  howpublished = {\url{https://github.com/wang-xinyu/tensorrtx}},
  commit = {668d89bbd5319c00be9f648d3db8f61689417d2d}
}
```

**Comment by MinnanZhou** [comment]
Got it!

**Comment by Hadimaamar** [comment]
> I successfully converted my trained model to a tensorrt engine based on your nice code on yolov5. The inference time with 416 size input is about 37ms.
> By the way, how to cite your code? I hope to use it in my thesis. Thanks a lot!

Heloo there iam realy struggling to make my own can you help me do it ,i can provide you with my best.pl file and the image files i realy need it for college

**Comment by MinnanZhou** [comment]
> > I successfully converted my trained model to a tensorrt engine based on your nice code on yolov5. The inference time with 416 size input is about 37ms.
> > By the way, how to cite your code? I hope to use it in my thesis. Thanks a lot!
> 
> Heloo there iam realy struggling to make my own can you help me do it ,i can provide you with my best.pl file and the image files i realy need it for college

I made that on a Jetson Nano with Jetpack 4.5.1. I just downloaded code and made and I had not changed any environment configurations . I think you may check your environment.

**Comment by ahhhxxx** [comment]
> I successfully converted my trained model to a tensorrt engine based on your nice code on yolov5. The inference time with 416 size input is about 37ms.
> By the way, how to cite your code? I hope to use it in my thesis. Thanks a lot!

Can I add you on QQ?I am also running in Nano. Why does the detection take a long time?
我可以加你QQ吗？我也是在nano 运行为什么检测的时间比较久？
an@an-desktop:~/Desktop/tensorrt/tensorrtx/yolov5/build$ sudo ./yolov5 -d yolov5s.engine ../samples
334ms
132ms

**Comment by MinnanZhou** [comment]
> > I successfully converted my trained model to a tensorrt engine based on your nice code on yolov5. The inference time with 416 size input is about 37ms.
> > By the way, how to cite your code? I hope to use it in my thesis. Thanks a lot!
> 
> Can I add you on QQ?I am also running in Nano. Why does the detection take a long time?
> 我可以加你QQ吗？我也是在nano 运行为什么检测的时间比较久？
> an@an-desktop:~/Desktop/tensorrt/tensorrtx/yolov5/build$ sudo ./yolov5 -d yolov5s.engine ../samples
> 334ms
> 132ms

我输入的图像分辨率是384*288，也许可以对比一下用pytorch的速度看看是否有提升

**Comment by nanhui69** [comment]
> ou on QQ?I am also running in Nano. Why does the detection tak
请问你现在这个问题解决了么？

**Comment by guyiyifeurach** [comment]
Hi, I run yolov5s on jetson nano with my own data. It can run on FP32 and FP16.
And with FP16, it inferenced cost about 84ms with 640 * 640 size using images in local directories.
But while in network communication inference, where receive data from remote client and process it into cv::Mat, the time is very big in first time.
To compare with 1st and later images, I run ``doInference`` 5 times with same image, written as below:

img1:
![image](https://user-images.githubusercontent.com/32059046/144799504-5ab60b26-5062-4ed2-b952-f52e8b49ef53.png)

``0th`` is the first time inference time, and ``1th`` is later.... Memcpy time I think is normal.

And the 5 times inference is written in img2:

![image](https://user-images.githubusercontent.com/32059046/144799920-b5ea625a-6406-4f4c-8228-5fb663c83103.png)

Can you give me some directions？@wang-xinyu Thank you!

**Comment by wang-xinyu** [comment]
@guyiyifeurach Try to let nano always running at the highest frequency.

**Comment by guyiyifeurach** [comment]
Thank you, you are right.



---Original---
From: "Wang ***@***.***&gt;
Date: Tue, Dec 7, 2021 11:29 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [wang-xinyu/tensorrtx] yolov5s implemented on Jetson nano (#569)




 
@guyiyifeurach Try to let nano always running at the highest frequency.
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
